plugin/src/main/java/io/jenkins/plugins/redis/RedisFingerprintStorageDescriptor.java

```diff
@@ -27,166 +27,36 @@
-
-    public void setSocketTimeout(int socketTimeout) {
-        this.socketTimeout = socketTimeout;
-    }
-
-    public String getCredentialsId() {
-        return credentialsId;
-    }
-
-    public void setCredentialsId(String credentialsId) {
-        this.credentialsId = credentialsId;
-    }
-
-    public @NonNull String getUsername() {
-        StandardUsernamePasswordCredentials credential = getCredential(credentialsId);
-        return getUsernameFromCredential(credential);
-    }
-
-    private @NonNull String getUsernameFromCredential(@CheckForNull StandardUsernamePasswordCredentials credential) {
-        if (credential == null) {
-            return "default";
-        }
-        String username = credential.getUsername();
-        if (username.equals("")) {
-            return "default";
-        }
-        return username;
-    }
-
-    public @NonNull String getPassword() {
-        StandardUsernamePasswordCredentials credential = getCredential(credentialsId);
-        return getPasswordFromCredential(credential);
-    }
-
-    private @NonNull String getPasswordFromCredential(@CheckForNull StandardUsernamePasswordCredentials credential) {
-        if (credential == null) {
-            return "";
-        }
-        return credential.getPassword().getPlainText();
-    }
-
-    private StandardUsernamePasswordCredentials getCredential(String id) {
-        StandardUsernamePasswordCredentials credential = null;
-        List<StandardUsernamePasswordCredentials> credentials = CredentialsProvider.lookupCredentials(
-                StandardUsernamePasswordCredentials.class, Jenkins.get(), ACL.SYSTEM, Collections.emptyList());
-        IdMatcher matcher = new IdMatcher(id);
-        for (StandardUsernamePasswordCredentials c : credentials) {
-            if (matcher.matches(c)) {
-                credential = c;
-            }
-        }
-        return credential;
-    }
+    public static final String DEFAULT_HOST = "localhost";
+    public static final int DEFAULT_PORT = 6379;
+    public static final int DEFAULT_DATABASE = 0;
+    public static final boolean DEFAULT_SSL = false;
+    public static final int DEFAULT_CONNECTION_TIMEOUT = 2000;
+    public static final int DEFAULT_SOCKET_TIMEOUT = 2000;
+    public static final String DEFAULT_CREDENTIALS_ID = "";
 
     @Override
-    public boolean configure(StaplerRequest req, JSONObject json) {
-        json = json.getJSONObject("redis");
-        setHost(json.getString("host"));
-        setPort(json.getInt("port"));
-        setDatabase(json.getInt("database"));
-        setSsl(json.getBoolean("ssl"));
-        setCredentialsId(json.getString("credentialsId"));
-        setConnectionTimeout(json.getInt(("connectionTimeout")));
-        setSocketTimeout(json.getInt(("socketTimeout")));
-        save();
-        RedisFingerprintStorage.get().createJedisPoolFromConfig();
-        return true;
+    public String getDisplayName() {
+        return "Redis Fingerprint Storage";
```

Comment:
Later we could make it localizable

Reply:
Could you kindly elaborate on this, I am not sure what this means exactly

Reply:
https://www.jenkins.io/doc/developer/internationalization/i18n-source-code/

Reply:
Done!